src/app/History.ts

```diff
@@ -74,4 +74,19 @@ export namespace HistoryManager {
       history.push(history.location.pathname + '?' + urlParams.toString());
     }
   };
+
+  /**
+   * setGraphNamespaceParam: properly replace the namespace section of the url
+   * without going to another page -- just rewrite the namespace part of the url
+   * Also: don't do this if we in a zoomed in sub graph view
```

Comment:
I don't think you want this comment in here.  Also, does it cost anything to do a replace if the namespace is not changing?   If so, perhaps we should check to see if the namespace in history.location.search is the same as what we are trying to set.

Reply:
it doesn't cost anything to do a replace because we are just changing the url and not refreshing the page.
Also see:

> NOTE:
>  We have to use the 'history.location' because we are calling
>    * this from inside 'getDerivedStateFromProps' which is 'static'
>    * and therefore doesn't have access to the 'context.router'